common/app/controllers/EmailSignupController.scala

```diff
@@ -71,6 +76,14 @@ class EmailFormService(wsClient: WSClient) extends LazyLogging with RemoteAddres
         .addHttpHeaders(headers: _*)
         .post(consentMailerPayload)
     }
+  }
+
+  private def serviceUrl(form: EmailForm): String = {
+    if (NewslettersRemoveConfirmationStep.isSwitchedOn && !form.emailConfirmation) {
+      return s"${Configuration.id.apiRoot}/consent-signup"
```

Comment:
I would say that in Scala the convention is to avoid early returns as it's not a very functional approach, and it makes the code more difficult to read and predict, this is only a preference and I'm sure some may disagree, but in this code base at least for consistency let's use an if/else here pleae